cpp/src/Ice/ConnectionI.cpp

```diff
@@ -2685,10 +2702,14 @@ Ice::ConnectionI::sendResponse(OutgoingResponse response, uint8_t compress)
 bool
 Ice::ConnectionI::initialize(SocketOperation operation)
 {
+    if (_connectTimeout > chrono::seconds::zero())
```

Comment:
Isn't it possible for `initialize` to be called multiple times?

Reply:
I think it's possible but so rare that we don't need to optimize for the case where it's called multiple times.

If it's called multiple times, we schedule multiple connect timer tasks.

Reply:
But isn't it an error to schedule a timeout, which is already scheduled?

Reply:
To be clear if we return false from

```
if (s != SocketOperationNone)
{
    _threadPool->update(shared_from_this(), operation, s);
    return false;
}
```

This method would be called again, and the timer schedule would throw

```
if (!inserted)
{
    throw std::invalid_argument("task is already scheduled");
}
```

Reply:
That's not the case before we're schedule a _different_ timer task each time. It's created by make_shared:
```cpp
_timer->schedule(make_shared<ConnectTimerTask>(shared_from_this()), _connectTimeout);
```